### ISSUE TITLE
  deadlock in put operation

### ISSUE THREAD

**Issue by JThink** [Huawei-Hadoop/hindex#45]
Is user region and index region balanced at the same time ? 

When put data into Hbase, it cause dead-lock issues, after track the code, i found in <code>HRegion.batchMutate</code> method.

``` java
    startRegionOperation();
    if (coprocessorHost != null) {
        coprocessorHost.postStartRegionOperation(); ------a
    }
    try {
        ......
    } finally {
        closeRegionOperation(); ----------b
        if (coprocessorHost != null) {
            coprocessorHost.postCloseRegionOperation();
        }
    }
  ......  
```

When execute 'a' step, the index region has not been banlanced to this server, it will throw `IOException` and do not execute 'b' step, then cause the deadlock when closing this region.

Could you check it?

**Comment by chrajeshbabu** [comment]
Hi This is same as 
https://github.com/Huawei-Hadoop/hindex/issues/26
Will fix now..Thanks for reporting.

**Comment by chrajeshbabu** [comment]
@JThink 
We have fixed #26 now. Can you try it with latest build. Thanks..

**Comment by JThink** [comment]
@chrajeshbabu 
Thank you!
